gr-fec/lib/reed-solomon/rstest.c

```diff
@@ -95,20 +94,8 @@ int main()
             }
             errs = exercise_char(handle, Tab[i].ntrials);
         } else {
-#ifdef ALL_VERSIONS
-            if ((handle = init_rs_int(Tab[i].symsize,
-                                      Tab[i].genpoly,
-                                      Tab[i].fcs,
-                                      Tab[i].prim,
-                                      Tab[i].nroots)) == NULL) {
-                printf("init_rs_int failed!\n");
-                continue;
-            }
-            errs = exercise_int(handle, Tab[i].ntrials);
-#else
             printf("init_rs_init support is not enabled\n");
             exit(1);
-#endif
```

Comment:
Is this proper or do we need to 
```c++
} else {
#if !defined(ALL_VERSIONS)
printf(…);
exit(1);
#endif
}
```

?

Reply:
You are right, I will fix it then rebuild.

Reply:
Now that we're removing the `_int` functions, we can also remove the test cases in `Tab[]` that have `symsize > 8`, since those are exclusively for the `_int` functions, and also get rid of the `if (Tab[i].symsize <= 8)` and all the `else`.

Reply:
what if the the is not `Tab[i].symsize <= 8`, then it will not implement the code in the else statement `else{
#if !defined(ALL_VERSIONS)
            printf("init_rs_init support is not enabled\n");
            exit(1);
#endif
        }`
      I dont think we should remove the `if (Tab[i].symsize <= 8)` and all the `else`.

Reply:
Just to be clear, my suggestion was as follows:

Amend
```c++
struct {
    int symsize;
    int genpoly;
    int fcs;
    int prim;
    int nroots;
    int ntrials;
} Tab[] = {
    { 2, 0x7, 1, 1, 1, 10 },
    { 3, 0xb, 1, 1, 2, 10 },
    { 4, 0x13, 1, 1, 4, 10 },
    { 5, 0x25, 1, 1, 6, 10 },
    { 6, 0x43, 1, 1, 8, 10 },
    { 7, 0x89, 1, 1, 10, 10 },
    { 8, 0x11d, 1, 1, 32, 10 },
    { 8, 0x187, 112, 11, 32, 10 }, /* Duplicates CCSDS codec */
#ifdef ALL_VESIONS
    { 9, 0x211, 1, 1, 32, 10 },
    { 10, 0x409, 1, 1, 32, 10 },
    { 11, 0x805, 1, 1, 32, 10 },
    { 12, 0x1053, 1, 1, 32, 5 },
    { 13, 0x201b, 1, 1, 32, 2 },
    { 14, 0x4443, 1, 1, 32, 1 },
    { 15, 0x8003, 1, 1, 32, 1 },
    { 16, 0x1100b, 1, 1, 32, 1 },
#endif
    { 0, 0, 0, 0, 0 },
};
```
to
```c++
struct {
    int symsize;
    int genpoly;
    int fcs;
    int prim;
    int nroots;
    int ntrials;
} Tab[] = {
    { 2, 0x7, 1, 1, 1, 10 },
    { 3, 0xb, 1, 1, 2, 10 },
    { 4, 0x13, 1, 1, 4, 10 },
    { 5, 0x25, 1, 1, 6, 10 },
    { 6, 0x43, 1, 1, 8, 10 },
    { 7, 0x89, 1, 1, 10, 10 },
    { 8, 0x11d, 1, 1, 32, 10 },
    { 8, 0x187, 112, 11, 32, 10 }, /* Duplicates CCSDS codec */
    { 0, 0, 0, 0, 0 },
};
```

Then remove the condition
```c++
if (Tab[i].symsize <= 8)
```
since now this will always be true, taking the code inside the `if` outside, so that it always runs. Finally, remove
```c++
        } else{
#if !defined(ALL_VERSIONS)
            printf("init_rs_init support is not enabled\n");
            exit(1);
#endif
        }
```

By the way, this also makes me realize that we aren't building the test with `-DALL_VERSIONS`, and maybe we should.